contributing/development-setup.md

```diff
@@ -51,7 +51,20 @@ For more explained information about plugin override check [Development Override
 - Make sure that the PR title follows [*Conventional Commits*](https://www.conventionalcommits.org/).
 - Add comments around your new code that explain what's happening.
 - Commit and push your changes to your branch then submit a pull request against the `master` branch.
-- A repo maintainer will review the your pull request, and may either request additional changes or merge the pull request.
+- A repo maintainer will review your pull request. 
+
+### Merging a Pull Request
+Due to security reasons, there are restrictions on how external contributions can be handled, especially concerning the use of repository secrets and running tests from forks.
+Our GitHub setup does not allow sharing of variables and secrets across forks made by external contributors. This is a necessary security measure to prevent unauthorized access and potential vulnerabilities.
+
+As external contributors cannot merge changes from their forks directly nor create branches from the master, the contribution process involves co-authoring. Here's what will happen:
+
+- We will cherry-pick your commit into a new branch in our main repository.
+- We will then create a new pull request from this branch, adding you as a co-author to ensure your contribution is not lost.
+- Our team will carry out any necessary testing and review on the new pull request. You are welcome to participate in this phase by commenting and making further suggestions.
+- Once approved, the pull request will be merged into the main branch.
+- **Exception**: 
+For minor changes that do not affect the core functionality of the project and do not require running tests, exceptions to the above process may be made at the discretion of the project maintainers.
```

Comment:
I think we shouldn't do any exception even if it looks a tiny change.  

For example this was apparently a very simple pr: https://github.com/mongodb/awscdk-resources-mongodbatlas/pull/298.
but thanks to have created a PR I was able to see that it was failing because doc has to be updated also in another file.

There is also the problem that this TF repo doesn't have merge queue. 

So all in all I think it's better not to do any exception and always create a PR in our side.

Reply:
no strong opinion, we do mention here that it will be on our discretion and it is called out if for a change we "do not require running tests", lmk I can update in a follow-up